src/passes/inlineAssemblyTransformer.ts

```diff
@@ -0,0 +1,159 @@
+import { ASTMapper } from '../ast/mapper';
+import { AST } from '../ast/ast';
+import {
+  createExpressionStatement,
+  createIdentifier,
+  createNumberLiteral,
+  createTuple,
+} from '../utils/nodeTemplates';
+
+import {
+  InlineAssembly,
+  YulNode,
+  Identifier,
+  ASTNode,
+  BinaryOperation,
+  Assignment,
+  Block,
+  Expression,
+  ExpressionStatement,
+  VariableDeclaration,
+  UncheckedBlock,
+  Literal,
+  Statement,
+} from 'solc-typed-ast';
+import assert from 'assert';
+import { getErrorMessage, WillNotSupportError } from '../utils/errors';
+
+const binaryOps: { [name: string]: string } = { add: '+', mul: '*', div: '/', sub: '-' };
+
+class YulVerifier {
+  errors: string[] = [];
+
+  verifyNode(node: YulNode) {
+    const methodName = `create${node.nodeType}`;
+    if (!(methodName in YulTransformer.prototype))
+      this.errors.push(`${node.nodeType} is not supported`);
+    const method = this[methodName as keyof YulVerifier] as (node: YulNode) => void;
+    if (method != undefined) return method.bind(this)(node);
```

Comment:
we should replace all != with their strict counterpart !==
you can add this rule https://eslint.org/docs/latest/rules/eqeqeq and make it error so that they are not added in the future